pycbc/strain/strain.py

```diff
@@ -1889,11 +1897,15 @@ def advance(self, blocksize, timeout=10):
         self.strain[len(self.strain) - csize + self.corruption:] = strain[:]
         self.strain.start_time += blocksize
 
+        # Apply autogating on the last 16 s of the buffer
+        autogating_duration = 16 * self.sample_rate
```

Comment:
I think the 16 here should be an option with a default. Apart from that I share Ian's worry about performance, but if we see that there is not an impact, this seems good go to

Reply:
Seems reasonable to have the duration be exposed as an option.